File: judge.py
import requests
import base64

#IP="10.231.181.196"
#PORT="2358"

#JUDGE0_URL = f"https://{IP}:{PORT}"
JUDGE0_URL = "https://judge.darlon.com.br"

# Código Python a ser executado
source_code = """
print(input())
"""

# Entrada padrão (stdin)
stdin = """Olá, Turma!"""

# Converter ambos para Base64
encoded_source = base64.b64encode(source_code.encode("utf-8")).decode("utf-8")
encoded_stdin  = base64.b64encode(stdin.encode("utf-8")).decode("utf-8")

# Payload
payload = {
    "source_code": encoded_source,
    "language_id": 71,  # Python 3.x
    "stdin": encoded_stdin,
    "base64_encoded": True,
    "wait": True
}

# Enviar submissão
response = requests.post(f"{JUDGE0_URL}/submissions?base64_encoded=true&wait=true", json=payload)
result = response.json()

# Decodificar saída (vem em Base64)
stdout = base64.b64decode(result.get("stdout") or "").decode("utf-8", errors="ignore")
stderr = base64.b64decode(result.get("stderr") or "").decode("utf-8", errors="ignore")

print("Status:", result["status"]["description"])
print("Saída padrão:", stdout)
print("Erros:", stderr)

File: judge.js
import fetch from "node-fetch"; // instale com: npm install node-fetch
import { Buffer } from "buffer";

const JUDGE0_URL = "http://judge.darlon.com.br";

// Código Python a ser executado
const sourceCode = `
print(input())

# val_a = int(input())
# val_b = int(input())
# val_soma - val_a = val_b
# print(f"{val_a} + {val_b} = {val_soma}")
`;

// Entrada padrão (stdin)
const stdin = `5
10`;

// Converter código e stdin para Base64
const encodedSource = Buffer.from(sourceCode, "utf-8").toString("base64");
const encodedStdin = Buffer.from(stdin, "utf-8").toString("base64");

// Montar payload
const payload = {
  source_code: encodedSource,
  language_id: 71, // Python 3.x
  stdin: encodedStdin,
  base64_encoded: true,
  wait: true
};

// Enviar submissão e mostrar resultado
async function main() {
  try {
    const response = await fetch(`${JUDGE0_URL}/submissions?base64_encoded=true&wait=true`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload)
    });

    const result = await response.json();

    // Decodificar saída e erros (que vêm em Base64)
    const stdout = result.stdout
      ? Buffer.from(result.stdout, "base64").toString("utf-8")
      : "";
    const stderr = result.stderr
      ? Buffer.from(result.stderr, "base64").toString("utf-8")
      : "";

    console.log("Status:", result.status.description);
    console.log("Saída padrão:", stdout);
    console.log("Erros:", stderr);
  } catch (error) {
    console.error("Erro:", error);
  }
}

main();
